sklearn/utils/validation.py

```diff
@@ -349,7 +389,8 @@ def _ensure_no_complex_data(array):
                          "{}\n".format(array))
 
 
-def check_array(array, accept_sparse=False, accept_large_sparse=True,
+@_deprecate_positional_args
+def check_array(array, *, accept_sparse=False, accept_large_sparse=True,
```

Comment:
Maybe put it after `accept_sparse` when it's `check_array(array, ['csr'])` it is fairly explicit. And since you had to change it in a couple of places here, it's likely donstream projects also would have to. No strong opinion though.

Reply:
thanks for the review

I have a slight preference for making it kwonly, mainly for consistency with `_validate_data` which de-facto forces all arguments to be kwonly as well. So making it kwonly here actually eases the transition to use `_validate_data`

Reply:
OK, fair enough. Thanks.